piptools/repositories/pypi.py

```diff
@@ -91,7 +91,7 @@ def find_best_match(self, ireq, prereleases=None):
         Returns a Version object that indicates the best match for the given
         InstallRequirement according to the external repository.
         """
-        if ireq.editable:
+        if ireq.editable or (ireq.link and not ireq.link.is_artifact):
```

Comment:
Seems like this `x.link and not x.link.is_artifact` is a recurring pattern. Can you make a helper function for that? I.e. `is_vcs_link(ireq)`?